serpentTools/tests/test_microxs.py

```diff
@@ -0,0 +1,182 @@
+from serpentTools.parsers import MicroXSReader
+from serpentTools.messages import SerpentToolsException
+
+
+class TestBadFile(unittest.TestCase):
+    """
+    Test bad files.
+
+    Tests:
+        1. test_noResults: file with no results
+    Raises SerpentToolsException
+    """
+
+    def test_noResults(self):
+        """Verify that the reader raises error when no results exist in the file"""
+        badFile = os.path.join(TEST_ROOT, 'bad_results_file.m')
+        with open(badFile, 'w') as badObj:
+            for _line in range(5):
+                badObj.write(str(_line))
+        badReader = MicroXSReader(badFile)
+        with self.assertRaises(SerpentToolsException):
+            badReader.read()
+        os.remove(badFile)
+
+class TestMicroXSReader(unittest.TestCase):
+    """
+    Class functionality tests for the micro xs reader.
+
+    Tests:
+
+        1. test_emptyAttributes:
+                        check that some results are
+                        collected.
+        2. test_univNum:
+                        check that the number of universes
+                        is correct.
+        3. test_fluxdata:
+                        Check that flux ratios and
+                        uncertainties are properly stored
+        4. test_fydata:
+                        Check that fission yields values
+                        are properly stored
+        5. test_getFY:
+                        Check the getFY method implemented
+                        to obtain fission yields for a specific
+                        parent, energy and fission product.
+            Raises SerpentToolsException:
+                        non-existing parent or fission product
+        6. test_getXS:
+                        Check the getXS method implemented
+                        to obtain cross-sections for a specific
+                        isotope and reaction. Returns empty list
+                        for non existing isotope and reaction
+            Raises SerpentToolsException:
+                        non-existing universe or wrong
+                        isotope formatting
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.file = os.path.join(TEST_ROOT, 'ref_mdx0.m')
+        with rc:
+            rc['microxs.getFY'] = True
+            rc['microxs.getXS'] = True
+            rc['microxs.getFlx'] = True
+            cls.reader = MicroXSReader(cls.file)
+            cls.reader.read()
+
+    def test_emptyAttributes(self):
+        """Verify that results are not all empty"""
+        mdxFile = os.path.join(TEST_ROOT, 'ref_mdx0_noXS.m')
+        with rc:
+            rc['microxs.getFY'] = False  # do not store fission yields
+            mdxReader = MicroXSReader(mdxFile)
+            with self.assertRaises(SerpentToolsException):
+                mdxReader.read()
+
+    def test_univNum(self):
+        """Test the expected number of universes."""
+        expectedUniv = {'0'}
+        expectedKeys = set(expectedUniv)
```

Comment:
You can directly declare `expectedKeys = {'0'}` to make it a set

Reply:
Changed to:

expectedKeys = set('0')